Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm

```diff
@@ -1333,4 +1333,92 @@ static void webProcessPoolHighDynamicRangeDidChangeCallback(CFNotificationCenter
 }
 #endif
 
+static AdditionalFonts additionalFonts(const Vector<URL>& fontURLs)
+{
+    AdditionalFonts additionalFonts;
+    for (auto& fontURL : fontURLs) {
+        auto sandboxExtensionHandle = SandboxExtension::createHandleWithoutResolvingPath(fontURL.fileSystemPath(), SandboxExtension::Type::ReadOnly);
+        additionalFonts.fontDataList.append({ fontURL, WTFMove(*sandboxExtensionHandle) });
+    }
+    return additionalFonts;
+}
+
+void WebProcessPool::registerUserInstalledFonts(WebProcessProxy& process)
+{
+    auto blockPtr = makeBlockPtr([weakThis = WeakPtr { *this }] {
+        NSString *userInstalledFontsPath = [NSHomeDirectory() stringByAppendingPathComponent:@"Library/Fonts"];
+
+        NSDirectoryEnumerator<NSString *> *enumerator = [NSFileManager.defaultManager enumeratorAtPath:userInstalledFontsPath];
+
+        Vector<URL> fontURLs;
+        for (NSString *font in enumerator) {
+            if ([font hasSuffix:@"ttf"] || [font hasSuffix:@"ttc"] || [font hasSuffix:@"otf"]) {
+                NSString *fontsPath = [userInstalledFontsPath stringByAppendingPathComponent:font];
+                auto fontURL = URL::fileURLWithFileSystemPath(String(fontsPath));
+                fontURLs.append(fontURL);
+                RELEASE_LOG(Process, "Registering font url %s", fontURL.string().utf8().data());
+            }
+        }
+
+        RunLoop::protectedMain()->dispatch([weakThis, fontURLs = WTFMove(fontURLs)] {
+            if (RefPtr protectedThis = weakThis.get()) {
+                for (auto& process : protectedThis->m_processes)
+                    process->send(Messages::WebProcess::RegisterAdditionalFonts(additionalFonts(fontURLs)), 0);
```

Comment:
I' guessing this is right but wanted to double check with you. Are we intentionally re-creating new SandboxExtensions for the same files for each WebProcess? I'm guessing these things get consumed and cannot be re-used for several processes but wanted to check.

Reply:
Yes, good catch, this is intended behavior. I haven't added an audit token to target specific WebContent processes, but will do that in the next patch.

Thanks for reviewing! I will address the comments in the upcoming patch.